### DISCUSSION TITLE
Are fixed-width widgets outside a horizontal box rendered?

### DISCUSSION THREAD

**Discussion by amano-kenji** [jtdaugherty/brick#435] [Q&A] [answered]
Think

```
hLimit 10 $ hBox [str "123456789012345", str "12345"]
```

What happens to `str "12345"`?

Is it erased after being rendered? Or, is it not rendered at all? Does hBox avoid evaluating `str "12345"`?

**Comment by jtdaugherty** [comment, marked as answer]
It is not rendered at all. As for whether it is evaluated, I don't know without looking closely at the code, but I don't think you need to worry about that and I wouldn't recommend writing code that depends on knowing whether it's evaluated. (If you're concerned about performance, this is another case where Haskell's lazy evaluation will help.)

Here's how it works: both the `str` widgets are `Fixed` in the horizontal dimension. The box renders the first one with 10 columns to consume. It uses them all and gets cropped. Then there are zero left, so the second one is not rendered.
